lib/index.ts

```diff
@@ -45,37 +45,39 @@ export function splitBuildStream(
 			next: () => void,
 		): void => {
 			// Find the build context that this file should belong to
-			const matchingTask = _.find(tasks, (task) => {
+			const matchingTasks = _.filter(tasks, (task) => {
 				if (task.external) {
 					return false;
 				}
 				return PathUtils.contains(task.context!, header.name);
 			});
 
-			if (matchingTask != null) {
+			if (matchingTasks.length === 0) {
+				Utils.drainStream(stream)
+				.then(() => {
+					next();
+					// return null here to keep bluebird happy
+					return null;
+				})
+				.catch((e) => reject(new TarError(e)));
+			} else {
 				const newHeader = header;
```

Comment:
Why do we create a new reference to header here?

Reply:
Because we change it on [this line](https://github.com/resin-io-modules/resin-multibuild/pull/11/files#diff-13b5b151431c7e7a17f273559ed212d5R67), not strictly necessary but I believe it shows intent a bit better.

Reply:
I've seen that but it looks like we think we copy it while we are only creating a reference. It was just a nit anyway.

Reply:
Ah I see what you're saying, that could be a bit confusing. Ok you've convinced me, will change.